src/pow.cpp

```diff
@@ -456,6 +456,20 @@ int Log::compare(const Basic &o) const
     return arg_->__cmp__(s);
 }
 
+RCP<const Basic> Log::subs(const map_basic_basic &subs_dict) const
+{
+    RCP<const Log> self = rcp_const_cast<Log>(rcp(this));
```

Comment:
This is fine. Later we should create a function or a macro, that returns `RCP<const X> self` for a class `X`. The reason is that this idiom is the only place, where Teuchos::RCP would not work as a drop in replacement (since Teuchos does not store the refcount in our class). However, there is a way to use Teuchos::RCP for this idiom too, one has to add some member variables into the class, perhaps a weak RCP pointer to itself, or something like that. This would be done in Debug mode.

I created #450 for this issue.